File: diagnose_payments.py

import asyncio
from supabase import create_client, Client
import os
from dotenv import load_dotenv

# Try to find env vars
load_dotenv()

url = os.environ.get("VITE_SUPABASE_URL")
key = os.environ.get("VITE_SUPABASE_PUBLISHABLE_KEY")

async def diagnose_payments():
    if not url or not key:
        print("Error: Supabase credentials not found in environment.")
        return

    supabase: Client = create_client(url, key)
    
    print("Fetching last 10 payment requests...")
    try:
        # Note: This uses the anon key, so it will be subject to RLS.
        # This is actually GOOD because it will show us what the RLS allows us to see.
        # But to see EVERYTHING for diagnosis, we really need the service role key.
        # Since I don't have that, I'll try to fetch and see what comes back.
        
        response = supabase.table("payment_requests").select("*, profiles(name, department)").order("created_at", desc=True).limit(10).execute()
        
        if response.data:
            print(f"Found {len(response.data)} requests:")
            for req in response.data:
                print(f"- ID: {req['id']} | Status: {req['status']} | Dept: {req['department']} | Purpose: {req['purpose']} | Amount: {req['amount']}")
        else:
            print("No requests found (or blocked by RLS).")
            
    except Exception as e:
        print(f"Error fetching data: {e}")

if __name__ == "__main__":
    asyncio.run(diagnose_payments())

File: diagnose_payments.js

import { createClient } from '@supabase/supabase-js';
import { readFileSync } from 'fs';

async function diagnose() {
  try {
    const envFile = readFileSync('.env', 'utf8');
    const lines = envFile.split(/\r?\n/);
    const env = {};
    for (const line of lines) {
      if (line && !line.startsWith('#')) {
        const [key, ...rest] = line.split('=');
        if (key && rest.length > 0) {
          let val = rest.join('=').trim();
          // Strip quotes
          if ((val.startsWith('"') && val.endsWith('"')) || (val.startsWith("'") && val.endsWith("'"))) {
            val = val.substring(1, val.length - 1);
          }
          env[key.trim()] = val;
        }
      }
    }
    
    const supabaseUrl = env.VITE_SUPABASE_URL;
    const supabaseKey = env.VITE_SUPABASE_PUBLISHABLE_KEY;

    if (!supabaseUrl || !supabaseKey) {
      console.error('Missing credentials. Check .env content.');
      return;
    }

    const supabase = createClient(supabaseUrl, supabaseKey);

    console.log('--- DIAGNOSING RECENT PAYMENTS ---');
    const { data, error } = await supabase
      .from('payment_requests')
      .select('id, status, department, purpose, amount, created_at')
      .order('created_at', { ascending: false })
      .limit(10);

    if (error) {
      console.error('Database Error:', error.message);
      return;
    }

    if (!data || data.length === 0) {
      console.log('No visible records. RLS is filtering results for the public key.');
      return;
    }

    console.log(`Visible to ANON key (${data.length}):`);
    data.forEach((req, i) => {
      console.log(`${i+1}. [${req.status}] ID: ${req.id.slice(0,8)} | Dept: ${req.department} | Amt: ₹${req.amount} | Created: ${req.created_at}`);
    });
  } catch (err) {
    console.error('Fatal Script Error:', err.message);
  }
}

diagnose();
